### ISSUE TITLE
Update SYCL `safe_min/max`

### ISSUE THREAD

**Issue by wavefunction91** [wavefunction91/ExchCXX#32]
Reported by @ajaypanyala

**Comment by wavefunction91** [comment]
Ping @abagusetty

**Comment by wavefunction91** [comment]
@ajaypanyala Can you provide the error message along with your build environment?

**Comment by abagusetty** [comment]
This is the error:
```bash
[ 78%] Building CXX object _deps/exchcxx-build/src/CMakeFiles/exchcxx.dir/libxc.cxx.o
cd /home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-build/src && /soft/restricted/CNDA/updates/2023.05.15.001/oneapi/compiler/eng-20230614/compiler/linux/bin/icpx -DEXCHCXX_HAS_CONFIG_H=1 -DUSING_Libxc -Dexchcxx_EXPORTS -I/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-src/include -I/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-build/include -I/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-src/src -I/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/libxc-src/src -I/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/libxc-build/src -I/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/libxc-build "-DOMPI_SKIP_MPICXX" -O2 -g -DNDEBUG -Wall -march=native  -march=sapphirerapids -mtune=sapphirerapids -ffast-math -fsycl -fsycl-default-sub-group-size 16 -fsycl-unnamed-lambda -fsycl-device-code-split=per_kernel -sycl-std=2020 --gcc-toolchain=/opt/cray/pe/gcc/12.2.0/snos -DOMPI_SKIP_MPICXX -DUSE_GAUXC -DUSE_DPCPP -DUSE_MEMKIND -std=c++17 -fPIC -Wall -Wextra -Wpedantic -Wnon-virtual-dtor -Werror=non-virtual-dtor -Wshadow -Werror=shadow -MD -MT _deps/exchcxx-build/src/CMakeFiles/exchcxx.dir/libxc.cxx.o -MF CMakeFiles/exchcxx.dir/libxc.cxx.o.d -o CMakeFiles/exchcxx.dir/libxc.cxx.o -c /home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-src/src/libxc.cxx
icpx: warning: overriding '-march=native' option with '-march=sapphirerapids' [-Woverriding-t-option]
icpx: warning: overriding '-march=native' option with '-march=sapphirerapids' [-Woverriding-t-option]
icpx: warning: overriding '-march=native' option with '-march=sapphirerapids' [-Woverriding-t-option]
In file included from /home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-src/src/libxc.cxx:48:
/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-src/include/exchcxx/impl/builtin/util.hpp:71:54: error: use of undeclared identifier 'sycl'
SAFE_CONSTEXPR_INLINE(F) safe_max(F x, F y) { return sycl::fmax(x,y); }
                                                     ^
/home/abagusetty/exachemdev_DLPNO_develop_10-17-2023/TAMM/build_dft/GauXC_External-prefix/src/GauXC_External-build/_deps/exchcxx-src/include/exchcxx/impl/builtin/util.hpp:73:54: error: use of undeclared identifier 'sycl'
SAFE_CONSTEXPR_INLINE(F) safe_min(F x, F y) { return sycl::fmin(x,y); }
                                                     ^
```

Not so surprisingly when replacing `sycl::` with `std::` the error is gone. On the other hand, the standalone build of ExchCXX is fine though. Also the `sycl::fmin` and `sycl::fmax` is guarded via `__SYCL_DEVICE_ONLY__` that ensures only the device-side kernels use the namespace.

**Comment by abagusetty** [comment]
This is the way `libcxx.cxx` is built standalone:
```bash
[ 96%] Building CXX object src/CMakeFiles/exchcxx.dir/libxc.cxx.o
cd /home/abagusetty/ExchCXX/build/src && /soft/restricted/CNDA/updates/2023.05.15.001/oneapi/compiler/eng-20230614/compiler/linux/bin/icpx -DEXCHCXX_HAS_CONFIG_H=1 -DUSING_Libxc -I/home/abagusetty/ExchCXX/include -I/home/abagusetty/ExchCXX/build/include -I/home/abagusetty/ExchCXX/src -I/home/abagusetty/ExchCXX/build/_deps/libxc-src/src -I/home/abagusetty/ExchCXX/build/_deps/libxc-build/src -I/home/abagusetty/ExchCXX/build/_deps/libxc-build -O3 -DNDEBUG -fno-sycl-id-queries-fit-in-int -fsycl-device-code-split=per_kernel -fno-sycl-early-optimizations -Wall -Wextra -Wpedantic -Wnon-virtual-dtor -Werror=non-virtual-dtor -Wshadow -Werror=shadow -fsycl -MD -MT src/CMakeFiles/exchcxx.dir/libxc.cxx.o -MF CMakeFiles/exchcxx.dir/libxc.cxx.o.d -o CMakeFiles/exchcxx.dir/libxc.cxx.o -c /home/abagusetty/ExchCXX/src/libxc.cxx
```

**Comment by wavefunction91** [comment]
Closed via #33
